spanish-es/groth16_es.md

```diff
@@ -0,0 +1,526 @@
+
+#### Pasos de configuración confiable
+
+Concretamente, la configuración confiable calcula lo siguiente:
+$$\begin{align*}
+\alpha,\beta,\tau &\leftarrow \text{escalares aleatorios}\\
+[\tau^{n-1}G_1,\tau^{n-2}G_1,\dots,\tau G_1,G_1] &\leftarrow \text{srs para } \mathbb{G}_1\\
+[\tau^{n-1}G_2,\tau^{n-2}G_2,\dots,\tau G_2,G_2] &\leftarrow \text{srs para } \mathbb{G}_2\\
+[\tau^{n-2}t(\tau),\tau^{n-3}t(\tau),\dots,\tau t(\tau),t(\tau)] &\leftarrow \text{srs para }h(\tau)t(\tau)\\
+[\Psi_1]_1 &= (\alpha v_1(\tau) + \beta u_1(\tau) + w_1(\tau))G_1\\
+[\Psi_2]_1 &= (\alpha v_2(\tau) + \beta u_2(\tau) + w_2(\tau))G_1\\
+&\vdots\\
+[\Psi_m]_1 &= (\alpha v_m(\tau) + \beta u_m(\tau) + w_m(\tau))G_1\\
+\end{align*}$$
+
+La configuración confiable publica
+
+$$([\alpha]_1,[\beta]_2,\text{srs}_{G_1},\text{srs}_{G_2}[\Psi_1]_1,[\Psi_2]_1,\dots,[\Psi_m]_1)$$
+
+#### Pasos del probador
+
+El probador calcula
+
+$$\begin{align*}
+[A]_1 &= [\alpha]_1 + \sum_{i=1}^m a_iu_i(\tau)\\
+[B]_2 &= [\beta]_2 + \sum_{i=1}^m a_iv_i(\tau)\\
+[C]_1 &= \sum_{i=1}^m a_i[\Psi_i]_1 + h(\tau)t(\tau)\\
+\end{align*}$$
+
+Tenga en cuenta que reemplazamos el Polinomio "problemático"
+
+$$=\sum_{i=1}^m a_i\boxed{(\alpha v_i(\tau)+\beta u_i(\tau) + w_i(\tau))}$$
+
+(el que contenía $\alpha$ y $\beta$) con
+
+$$\sum_{i=1}^m a_i[\Psi_i]_1$$
+
+#### Pasos del verificador
+
+El verificador calcula:
+
+$$[A]_1\bullet[B]_2 \stackrel{?}= [\alpha]_1 \bullet [\beta]_2 + [C]_1\bullet G_2$$
+
+## Compatibilidad con entradas públicas
+
+Hasta ahora, la fórmula del verificador no admite entradas públicas, es decir, hacer pública una parte del testigo.
+
+Por convención, las partes públicas del testigo son los primeros $\ell$ elementos del vector $\mathbf{a}$. Para hacer públicos esos elementos, el verificador simplemente los revela:
+
+$$[a_1, a_2, \dots, a_\ell]$$
+
+Para que el verificador pruebe que esos valores se utilizaron de hecho, debe realizar algunos de los cálculos que el verificador estaba haciendo originalmente.
+
+En concreto, el probador calcula:
+
+$$\begin{align*}
+[A]_1 &= [\alpha]_1 + \sum_{i=1}^m a_iu_i(\tau)\\
+[B]_2 &= [\beta]_2 + \sum_{i=1}^m a_iv_i(\tau)\\
+[C]_1 &= \sum_{i=\ell+1}^m a_i[\Psi_i]_1 + h(\tau)t(\tau)\\
+\end{align*}$$
+
+Obsérvese que solo cambió el cálculo de $[C]_1$: el probador solo utiliza los términos $a_i$ y $\Psi_i$ $\ell + 1$ a $m$.
+
+El verificador calcula los primeros términos $\ell$ de la suma:
+$$[X]_1=\sum_{i=1}^\ell a_i\Psi_i$$
+
+Y la ecuación de verificación es:
+
+$$[A]_1\bullet[B]_2 \stackrel{?}= [\alpha]_1 \bullet [\beta]_2 + [X]_1\bullet G_2 + [C]_1\bullet G_2$$
+
+## Parte 2: Separación de las entradas públicas de las entradas privadas con $\gamma$ y $\delta$
+
+La suposición en la ecuación anterior es que el probador solo está usando $\Psi_{\ell+1}$ a $\Psi_m$ para calcular $[C]_1$, pero nada impide que un probador deshonesto use $\Psi_1$ a $\Psi_{\ell}$ para calcular $[C]_1$, lo que posiblemente lleve a una prueba falsificada.
+

```

Comment:
missed some paragraphs after this sentence

"""
Por ejemplo, aquí está nuestra ecuación de verificación actual:

$$[A]_1\bullet[B]_2 \stackrel{?}= [\alpha]_1 \bullet [\beta]_2 + \sum_{i=1}^\ell a_i\Psi_i +  [C]_1\bullet G_2$$

Si ampliamos el término C bajo el capó, obtenemos lo siguiente:

$$[A]_1\bullet[B]_2 \stackrel{?}= [\alpha]_1 \bullet [\beta]_2 + \sum_{i=1}^\ell a_i\Psi_i + \underbrace{(\sum_{i=\ell+1}^m a_i[\Psi_i]_1 + h(\tau)t(\tau))}_C \bullet G_2$$

Supongamos por ejemplo y sin pérdida de generalidad que $\mathbf{a} = [1,2,3,4,5]$ y $\ell=3$. En ese caso, la parte pública del testigo es $[1,2,3]$ y la parte privada es $[4,5]$.

La ecuación final sería la siguiente:

$$[A]_1\bullet[B]_2 \stackrel{?}= [\alpha]_1 \bullet [\beta]_2 + (1\Psi_1+2\Psi_2+3\Psi_3)\bullet G2 + \underbrace{(4\Psi_4 + 5\Psi_5  + h(\tau)t(\tau))}_C \bullet G_2$$

Sin embargo, nada impide que el probador cree una parte válida del testigo público como [1,2,0] y mueva la parte pública eliminada a cero a la parte privada del cálculo de la siguiente manera:

$$[A]_1\bullet[B]_2 \stackrel{?}= [\alpha]_1 \bullet [\beta]_2 + (1\Psi_1+2\Psi_2+\boxed{0\Psi_3})\bullet G2 + \underbrace{(\boxed{3\Psi_3}+4\Psi_4 + 5\Psi_5  + h(\tau)t(\tau))}_C \bullet G_2$$

La ecuación anterior es válida, pero el testigo no necesariamente satisface las restricciones originales.

Por lo tanto, debemos evitar que el probador use $\Psi_1$ a $\Psi_{\ell}$ como parte del cálculo de $[C]_1$.


### Presentamos $\gamma$ y $\delta$
"""